app/Http/Resources/AdminMessageResource.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\Http\Resources;
+
+use Illuminate\Http\Resources\Json\JsonResource;
+use App\Models\User;
+
+class AdminMessageResource extends JsonResource
+{
+    /**
+     * Transform the resource into an array.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return array|\Illuminate\Contracts\Support\Arrayable|\JsonSerializable
+     */
+    public function toArray($request)
+    {
+        return [
+            'id' => $this->id,
+            'created_at' => $this->created_at->toDateTimeString(),
+            'sender_id' => $this->sender_id,
+            'sender_username' => User::find($this->sender_id)->username,
```

Comment:
You can change this to $this->sender->username, although realistically it does the same thing.

Reply:
Maybe it's better to do `User::find($this->sender_id)->value('username')`

Reply:
changed to $this->sender->value('username')